src/context/WalletProvider/SelectModal.tsx

```diff
@@ -12,26 +17,29 @@ export const SelectModal = ({ connect }: { connect: IWalletContext['connect'] })
       <ModalBody>
         <Text mb={6} color='gray.500' translation={'walletProvider.selectModal.body'} />
         <Stack mb={6}>
-          {Object.keys(SUPPORTED_WALLETS).map(key => {
-            const option = SUPPORTED_WALLETS[key]
-            const Icon = option.icon
-            return (
-              <Button
-                variant='ghost-filled'
-                colorScheme='blue'
-                key={key}
-                w='full'
-                h='auto'
-                px={6}
-                py={4}
-                justifyContent='space-between'
-                onClick={() => connect(key)}
-              >
-                <Text fontWeight='semibold'>{option.name}</Text>
-                <Icon height='auto' w='45px' />
-              </Button>
-            )
-          })}
+          {adapters &&
+            // Only iterate over the adapters that successfully initialized
```

Comment:
we should probably render supported but unavailable wallets differently with a tooptip or similar giving the user feedback as to why a plugged in keepkey might be unavailable, but we can do that in a separate PR

Reply:
I can revert this change.